### PR TITLE
Linux support

### PR DESCRIPTION
**Pull request by misterhat**
this should use the distro's install of ffmpeg before downloading any binaries off of npmjs

### CODE REVIEW

**Comment by misterhat** [comment]
didn't realize there was already a PR for this, but mine seems a bit simpler. i think it should check for win32 before linux, and then use ffmpeg as the last fallback for other UNIXes.

**Comment by PaulBGD** [comment]
I like it, it is simple. One of the advantages mine has however is that the bash script accepts the EXACT arguments that the mac binary accepts. This means that we don't have to do any special parsing of code (like you do in the JS code.)

**Comment by dmitry** [comment]
I'd say it would be great to support `avconv` as well (latest changelog: https://git.libav.org/?p=libav.git;a=blob;f=Changelog;hb=refs/tags/v12).

**Comment by misterhat** [comment]
@dmitry agreed, but it's not packaged in my distribution. I think the arguments should be almost identical.

**Comment by dmitry** [comment]
@misterhat could you please include it. Could be checked via `which ffmpeg` - if found, then use it, in other case `which avconv` could be used as an alternative way. Then I can check how it works, so I'm :up: for a beta tester.

**Comment by WorldStarCoder** [comment]
I think we should stick with ffmpeg,,,friend

We really need this patch ASAP, its very important,,,for project sussess THX

File: index.js [hunk 1]
```diff
@@ -34,7 +34,22 @@ class Aperture {
...
 
-      this.recorder = execa(path.join(__dirname, 'swift', 'main'), recorderOpts);
+      if (process.platform === 'darwin') {
+        this.recorder = execa(path.join(__dirname, 'swift', 'main'), recorderOpts);
+      } else if (process.platform === 'linux') {
+        let args = ['-f', 'x11grab', '-i'];
+
+        if (opts.cropArea) {
+          args.push(':0+' + opts.cropArea.x + ',' + opts.cropArea.y);
+          args.push('-video_size', opts.cropArea.width + 'x' + opts.cropArea.height);
```
Comment (sindresorhus) [inline index.js]: Use template literals for these

File: index.js [hunk 2]
```diff
@@ -34,7 +34,22 @@ class Aperture {
         recorderOpts.push(`${cropArea.x}:${cropArea.y}:${cropArea.width}:${cropArea.height}`);
       }
 
-      this.recorder = execa(path.join(__dirname, 'swift', 'main'), recorderOpts);
+      if (process.platform === 'darwin') {
+        this.recorder = execa(path.join(__dirname, 'swift', 'main'), recorderOpts);
+      } else if (process.platform === 'linux') {
+        let args = ['-f', 'x11grab', '-i'];
```
Comment (sindresorhus) [inline index.js]: `let` => `const`

And use long flags whenever possible.

File: package.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
   "main": "index.js",
   "scripts": {
     "test": "xo",
-    "build": "cd swift && xcodebuild && mv build/release/aperture main && rm -r build",
-    "postinstall": "npm run build"
+    "build-macos": "cd swift && xcodebuild && mv build/release/aperture main && rm -r build",
+    "postinstall": "case $OSTYPE in darwin*) npm run build-macos ;; esac"
```
Comment (sindresorhus) [inline package.json]: I think `[[ $OSTYPE == darwin* ]] && npm run build-macos` would be simpler.

**Comment by sindresorhus** [comment]
Can you also include support for the `draw_mouse` option? In master we have an option called `showCursor`. https://ffmpeg.org/ffmpeg-devices.html#x11grab

File: package.json (hunk 3 above)
Comment (misterhat) [inline package.json]: It would be, but it's not supported in plain shell.

Comment (sindresorhus) [inline package.json]: I don't think we care about that, but you easily do `[ "$(uname)" = "Darwin" ] && npm run build-macos`  then.

**Comment by sindresorhus** [comment]
@misterhat Hey. Any interest in finishing this up? :)

**Comment by zedtux** [comment]
@misterhat +1

**Comment by fearphage** [comment]
> Could be checked via `which ffmpeg`

FYI, don't use `which` in scripts to check for existence. The exit code is not reliable. Use `command`:

```shell 
if command -v ffmeg > /dev/null 2>&1; then 
  # do work 
fi;
```

Are these changes abandoned? I wouldn't mind wrapping them up if needed.

**Comment by pluma** [comment]
I guess the PR has been abandoned. Are the changes that were requested all that is needed to get Linux support working in aperture (and thus kap)? I would be willing to fork @misterhat's PR and make the requested changes myself.

**Comment by PaulBGD** [comment]
@pluma I had previously created a working version of kap + aperture that ran on linux. It's a bit outdated, but it exists on my profile with support for ubuntu/centos/arch.

**Comment by sindresorhus** [comment]
Closing in favor of #32

**Pull request closed without merging**